### PR TITLE
Rename ready callback

### PR DESCRIPTION
**Pull request by popomore**

### CODE REVIEW

**Comment by popomore** [comment]
@fengmk2  I renamed this repo, should I transfer it out of koa?

File: lib/ready.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+   *   - {Boolean} isWeakDep - whether it's a weak dependency, default: false
+   */
+  constructor(opt) {
+    super();
+    this.opt = opt || {};
+    this.isError = false;
+    this.cache = new Map();
+    ready.mixin(this);
+    setImmediate(function () {
+      // 如果没有任何任务, 则触发 ready
```
Comment (fengmk2) [inline lib/ready.js]: use en?

**Comment by fengmk2** [comment]
mv to node-modules

File: lib/ready.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const EventEmitter = require('events');
+const once = require('once');
+const ready = require('ready');
```
Comment (fengmk2) [inline lib/ready.js]: use get-ready instead? https://github.com/node-modules/ready

File: package.json [hunk 3]
```diff
@@ -9,30 +9,31 @@
...
   "scripts": {
-    "lint": "jshint .",
-    "test": "_mocha -R spec -t 20000",
-    "cov": "jshint . && node --harmony node_modules/istanbul/lib/cli.js cover node_modules/mocha/bin/_mocha -- -R spec -t 20000",
-    "coveralls": "cat ./coverage/lcov.info | coveralls",
+    "lint": "eslint .",
+    "test": "mocha -R spec -t 20000",
+    "cov": "npm run lint && istanbul cover --report=none node_modules/mocha/bin/_mocha -- -R spec -t 20000 && istanbul report json html",
     "autod": "autod -e test/fixtures -f ~ -w"
   }
```
Comment (fengmk2) [inline package.json:41]: add engines

``` js
"engines": {
  "node": ">=2.0.0"
} 
```

**Comment by popomore** [comment]
Done

File: lib/ready.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+   * @param  {String} name -
+   * @param  {Object} opt - the options that will override global
+   * @return {Function} - a callback
+   */
+  readyCallback(name, opt) {
+    const id = uuid.v1();
+    opt = extend({}, defaults, this.opt, opt);
+    opt.name = name || id;
+    const timer = setTimeout(this.emit.bind(this, 'ready_timeout', opt.name), opt.timeout);
+    const cb = once(function(err) {
```
Comment (fengmk2) [inline lib/ready.js:73]: ~~Why need `once`? I don't see `cb` will call multi times anymore.~~

Comment (fengmk2) [inline lib/ready.js:73]: oh...I got it.

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
`ready-callback@1.0.0`

**Pull request merged**